File: Find All Numbers Disappeared in an Array/FindAllNumbersDisappearedinanArray.java
import java.util.*;

public class FindAllNumbersDisappearedinanArray {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> ret = new LinkedList<Integer>();
        if(nums == null || nums.length == 0) return ret;
        
        for(int i=0;i<nums.length;i++){
        	int index = Math.abs(nums[i]) - 1;
        	if(nums[index] > 0){
        		nums[index] = -nums[index];
        	}
        }
        
        for(int i=0;i<nums.length;i++){
        	if(nums[i] >0){
        		ret.add(i+1);
        	}
        }
        
        return ret;
    }
}

File: Find All Numbers Disappeared in an Array/FindAllNumbersDisappearedinanArray.py
class FindAllNumbersDisappearedinanArray(object):
    def findDisappearedNumbers(self, nums):
        ret = list()
        if nums is None or len(nums)==0:
            return ret
        for i in range(0,len(nums)):
            index = abs(nums[i]) - 1
            if nums[index] > 0:
                nums[index] = -nums[index]
        for i in range(0,len(nums)):
            if nums[i] > 0:
                ret.append(i+1)
        return ret
                
        